.github/workflows/rust.yml

```diff
@@ -27,6 +43,11 @@ jobs:
         with:
           path: /home/runner/.cargo
           key: cargo-cache
+      - name: Optionally update upstream dependencies
+        if: github.event_name == 'schedule' || needs.detect-ci-trigger.outputs.triggered == 'true'
```

Comment:
Do we want to run this workflow on a schedule? If not, we can remove the relevant condition:

```suggestion
        if: needs.detect-ci-trigger.outputs.triggered == 'true'
```

Reply:
I was planning on adding a schedule to this one as well, though I forgot to add it. Given that the other test will check nightly df as well we can probably skip the schedule here